src/lkl/setup.c

```diff
@@ -402,28 +384,42 @@ static void* lkl_activate_crypto_disk_thread(struct lkl_crypt_device* lkl_cd)
 
     crypt_free(cd);
 
-    // The key is only needed during activation, so don't keep it around
-    // afterwards and free up space.
+    // The key is only needed during activation, so clear it now.
     memset(lkl_cd->disk_config->key, 0, lkl_cd->disk_config->key_len);
-
-    unsigned long munmap_ret;
-    if ((munmap_ret = lkl_sys_munmap(
-             (unsigned long)lkl_cd->disk_config->key,
-             lkl_cd->disk_config->key_len)))
-    {
-        sgxlkl_fail(
-            "Unable to unmap memory for disk encryption key: %s\n",
-            lkl_strerror((int)munmap_ret));
-    }
     lkl_cd->disk_config->key = NULL;
-    lkl_cd->disk_config->key_len = 0;
```

Comment:
Why is key_len no longer set to 0 when its corresponding key is nulled?

Reply:
Thank you! I re-added the line. My guess is that this code is an attempt to scrub the key since it is no longer needed. I assumed the size is not a secret, but I added it back just in case something uses it later. :)